### ISSUE TITLE
what is the correct way to monitor RealTime Database changes using the plugin?

### ISSUE THREAD

**Issue by carlosdelfino** [EddyVerbruggen/nativescript-plugin-firebase#1392]
I wrote the code below to monitor the "/ users" path in Realtime Database, so that whenever a new user is added to this path the application is informed, but I only get data when the code is initialized when the listener is added, ie only once returns all the data, I understand this initial operation, but what is preventing the callback call when new data is added or when some data is changed or removed?

```
    async init() {
        this._settings.debug && console.log("PersistenceService.init");
        return await firebase.init({ persist: true })
            .then((result) => {
                if (this._settings.debug)
                    console.log("PersistenceService.init firebase.init.then():", result);
                firebase.keepInSync("/users", true).then(() => {
                    var onValueEvent = (result: firebase.FBData) => {
                        console.log("Event type: " + result.type);
                        console.log("Key: " + result.key);
                        console.log("Value: " + JSON.stringify(result.value));
                    };
                    firebase.addChildEventListener(onValueEvent, "/users").then(
                        (listenerWrapper: firebase.AddEventListenerResult) => {
                            this._settings.debug && console.log("PersistenceService.init firebase.init.then firebase.addChildEventListener");
                            var path = listenerWrapper.path;
                            var listeners = listenerWrapper.listeners; // an Array of listeners added
                            // you can store the wrapper somewhere to later call 'removeEventListeners'
                        }
                    ).catch((error) => console.log("PersistenceService.init firebase.init.then firebase.addChildEventListener.catch", error));
                });

                return result;
            })
            .catch((error) => {
                if (this._settings.debug)
                    console.log("AuthService.new(): firebase.init.catch()", error);
                throw error;
            })
    }
```

The codes below work perfectly:

```
   /** */
    registerStatus(user: IslamicUser): void {
        firebase.setValue("/users/" + user.uid, user);
    }

    /** */
    updateUserPosition(user: IslamicUser, position: GPSInfo) {
        if (this._settings.debug)
            console.log("PersistenceService.updateUserPosition", user, position);
        position.updated = ServerValue.TIMESTAMP;
        firebase.setValue("/users/" + user.uid + "/position", position)
            .then(() => {
                if (this.debug) console.log("PersistenceService.updateUserPosition firebase.setValue");
                return firebase.getValue("/users/" + user.uid + "/position/updated")
                    .then((snapshot) => {
                        if (this._settings.debug)
                            console.log("PersistenceService.updateUserPosition firebase.getValue", snapshot);
                        return firebase
                            .setValue("/users_position_history/" + user.uid + "/posistions/" + snapshot.value, position);
                    })
            });
    }
}
```
thanks

**Comment by carlosdelfino** [comment]
Hi @EddyVerbruggen Is this a bug or am I doing it wrong? What would be the correct way to monitor child paths in RealTime Database using the plugin?

**Comment by carlosdelfino** [comment]
@EddyVerbruggen, I found the cause of the error.
When adding the listener even rules not allowing access to the path, in the first instant the data update process the listener runs as "ChildAdded", then presents an error, which I was not capturing, which returned the data as null.

Thereafter it no longer runs the listener even though the user is already authenticated and access to the path has been granted.

For this to work, the listener must be added only when user access is allowed to the path to be monitored.

It would be interesting to add such working details to the Realtime Database tutorial.

I found it very strange that it allows data access, even without authorization at startup, in this case is a failure of the security system Firebase Realtime Database.
